pkg/webhook/controlplane/ensurer.go

```diff
@@ -125,27 +127,35 @@ func (e *ensurer) EnsureKubeletServiceUnitOptions(ctx context.Context, gctx gcon
 	return new, nil
 }
 
-func ensureKubeletCommandLineArgs(command []string) []string {
+func ensureKubeletCommandLineArgs(command []string, kubeletVersion *semver.Version) []string {
 	// TODO: Figure out how to provide the provider-id via the kubelet config file (as of Kubernetes 1.19 the kubelet config
 	// offers a new `providerID` field which can be used, and it's expected that `--provider-id` will be deprecated eventually).
 	// Today, the problem is that the provider ID is determined dynamically using the script above, but the kubelet config cannot
 	// reference environment variables like it's possible today with the CLI parameters.
 	// See https://github.com/kubernetes/kubernetes/pull/90494
 	command = extensionswebhook.EnsureStringWithPrefix(command, "--provider-id=", "${PROVIDER_ID}")
```

Comment:
Do we need to move `provider-id` to config file too?

Reply:
Eventually yes, but not part of this PR. I have talked about this in https://github.com/gardener/gardener-extension-provider-alicloud/issues/202#issuecomment-1022443917

Reply:
> Do we need to move `provider-id` to config file too?

It is now addressed in TODO. For now I think we can keep it this way and figure out a solution later.